src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,130 @@
+            Todo create(String[] line) {
+                return new Todo(line[2], Boolean.parseBoolean(line[1]));
+            }
+
+            @Override
+            void check(String[] line) throws DukeException {
+                if (line.length != 3) {
+                    throw new DukeException("The file format is wrong :((.");
+                }
+            }
+        },
+        D {
+            @Override
+            Deadline create(String[] line) {
+                try {
+                    return new Deadline(line[2], LocalDate.parse(line[3]),
+                            Boolean.parseBoolean(line[1]));
+                } catch (Exception e) {
+                    throw e;
+                }
+            }
+        },
+        E {
+            @Override
+            Event create(String[] line) {
+                try {
+                    return new Event(line[2], LocalDate.parse(line[3]),
+                            Boolean.parseBoolean(line[1]));
+                } catch (Exception e) {
+                    throw e;
+                }
+            }
+        };
+
+        abstract Task create(String[] line);
+
+        void check(String[] line) throws DukeException {
+            if (line.length != 4) {
+                throw new DukeException("The file format is wrong :((.");
+            }
+        }
+    }
+
+    /**
+     * Storage constructor.
+     *
+     * @param filePath path of the the storage file.
+     */
+    public Storage(String filePath) throws DukeException {
+        file = new File("./data/dukeFile.txt");
+        file.getParentFile().mkdirs();
+
+        try {
+            file.createNewFile();
+        } catch (Exception e) {
+            throw new DukeException("Can't create file .__.");
+        }
+    }
+
+    /**
+     * Reads data from the file and converts it to a list of tasks.
+     *
+     * @return a list of tasks.
+     * @throws DukeException can't read or convert the file to a list of tasks.
+     */
+    public ArrayList<Task> readFromFile() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner fileIn = null;
+        try {
+            fileIn = new Scanner(file);
+
+            while (fileIn.hasNextLine()) {
+                String[] line = fileIn.nextLine().split(" \\| ");
+
+                TaskType taskType;
+                try {
+                    taskType = TaskType.valueOf(line[0]);
+                    taskType.check(line);
+                    tasks.add(taskType.create(line));
+                } catch (Exception e) {
+                    throw new DukeException("The file format is wrong :((.");
+                }
+            }
+
+            return tasks;
+        } catch (FileNotFoundException e) {
+            throw new DukeException("The file is not found TwT");
+        } catch (Exception e) {
+            throw e;
+        } finally {
+            if (fileIn != null) {
+                fileIn.close();
+            }
+        }
```

Comment:
Instead of using nested try-catch blocks, would it be better to use one block to catch different kinds of exceptions and throw the corresponding DukeExceptions?